prevent the compiler from optimizing out the 'this' pointer check

From: Christian Prochaska <christian.prochaska@genode-labs.com>


---
 include/kobject.hpp |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/kobject.hpp b/include/kobject.hpp
index b309401..7f089d3 100644
--- a/include/kobject.hpp
+++ b/include/kobject.hpp
@@ -46,6 +46,9 @@ class Kobject : public Mdb
         explicit Kobject (Type t, Space *s, mword b = 0, mword a = 0, void (*f)(Rcu_elem *) = free, void (*pref)(Rcu_elem *) = nullptr ) : Mdb (s, reinterpret_cast<mword>(this), b, a, f, pref), objtype (t) {}
 
     public:
+
+        /* prevent the compiler from optimizing out the 'this' pointer check */
+        __attribute__((optimize("-fno-delete-null-pointer-checks")))
         ALWAYS_INLINE
         inline Type type() const
         {
